etna/models/nn/tft_native/layers.py

```diff
@@ -0,0 +1,293 @@
+        """
+        x = self.glu(x)
+        x = self.norm(x + residual)
+        return x
+
+
+class GatedResidualNetwork(nn.Module):
+    """Gated Residual Network (GRN)."""
+
+    def __init__(self, input_size: int, output_size: int, dropout: float = 0.1, context: bool = False) -> None:
+        """Init GRN.
+
+        Parameters
+        ----------
+        input_size:
+            input size of the feature representation
+        output_size:
+            output size of the feature representation
+        dropout:
+            dropout rate
+        context:
+            whether to pass context vector through the block
+        """
+        super().__init__()
+        self.input_size = input_size
+        self.output_size = output_size
+        self.dropout = dropout
+        self.context = context
+
+        self.fc1 = nn.Linear(self.input_size, self.input_size)
+        self.elu = nn.ELU()
+
+        if self.context:
+            self.context_fc = nn.Linear(self.input_size, self.input_size, bias=False)
+
+        self.residual_fc = nn.Linear(self.input_size, self.output_size) if self.input_size != self.output_size else None
+        self.fc2 = nn.Linear(self.input_size, self.input_size)
+
+        self.gate_norm = GateAddNorm(
+            input_size=self.input_size,
+            output_size=self.output_size,
+            dropout=self.dropout,
+        )
+
+    def forward(self, x: torch.Tensor, context: Optional[torch.Tensor] = None) -> torch.Tensor:
+        """Forward pass.
+
+        Parameters
+        ----------
+        x:
+            batch of data with shapes (batch_size, num_timestamps, input_size)
+        context:
+            batch of data passed as the context through the block with shapes (batch_size, num_timestamps, output_size)
+        Returns
+        -------
+        :
+            output batch of data with shapes (batch_size, num_timestamps, output_size)
+        """
+        residual = self.residual_fc(x) if self.residual_fc is not None else x
+        x = self.fc1(x)
+        if context is not None:
+            x = x + self.context_fc(context)
+        x = self.elu(x)
+        x = self.fc2(x)
+        x = self.gate_norm(x=x, residual=residual)
+        return x
+
+
+class VariableSelectionNetwork(nn.Module):
+    """Variable Selection Network."""
+
+    def __init__(self, input_size: int, features: Tuple[str], context: bool = False, dropout: float = 0.1) -> None:
+        """Init Variable Selection Network.
+
+        Parameters
+        ----------
+        input_size:
+            input size of the feature representation
+        features:
+            features to pass through the block
+        context:
+            whether to pass context vector through the block
+        dropout:
+            dropout rate
+        """
+        super().__init__()
+        self.input_size = input_size
+        self.features = features
+        self.context = context
+        self.dropout = dropout
+        self.grns = {
```

Comment:
It should be [`ModuleDict`](https://pytorch.org/docs/stable/generated/torch.nn.ModuleDict.html) -- otherwise modules won't be properly registered